drift-codec/src/main/java/com/facebook/drift/codec/internal/ProtocolReader.java

```diff
@@ -127,6 +128,18 @@ public ByteBuffer readBinaryField()
         return fieldValue;
     }
 
+    public ByteBuffer readBinaryField(byte[] buf, int offset, int length)
+            throws TException
+    {
+        if (!checkReadState(TType.STRING)) {
+            return null;
+        }
+        currentField = null;
+        ByteBuffer fieldValue = ((TBinaryProtocol) protocol).readBinary(buf, offset, length);
```

Comment:
Let's don't downcast it. If you don't want to implement it for other protocols - you can leave it as `throw UnsupportedOperationException`. But basically keep it virtual.

Reply:
I implemented it in other protocols as well. So removed the cast.